gsoc/gsoc2016/ideas_yt.md

```diff
@@ -0,0 +1,166 @@
+engineering simulation codes. It can handle data produced by particle-based
+codes, as well data produced by codes that use various types of mesh structures,
+including uniform and adaptively refined meshes as well as unstructured and
+semistructured meshes.
+
+To learn more about how to use yt to interact with simulation data, take a look
+at the [quickstart guide](http://yt-project.org/doc/quickstart/index.html), as
+well as the rest of [the yt documentation](http://yt-project.org/doc). We also
+provide [a listing of sample test datasets](http://yt-project.org/data) that can
+be loaded by yt.
+
+For more information about contributing to yt, take a look at our
+[developer guide](http://yt-project.org/docs/dev/developing/index.html). To see
+discussions about past yt projects, take a look at the yt enhancement proposal
+(YTEP) [listing](https://ytep.readthedocs.org).
+
+### Integrate yt plots with interactive matplotlib backends
+
+*Suggested Mentor(s):* [Nathan Goldbaum](https://bitbucket.org/ngoldbaum),
+ [Matthew Turk](https://bitbucket.org/MatthewTurk/)
+
+*Difficulty:* Intermediate
+
+*Knowledge needed:* Familiarity with matplotlib. Knowledge of matplotlib's
+ object oriented API a plus.
+
+*Programming skills:* Python. GUI programming.
+
+#### Description
+
+Currently, all yt plotting objects have a `show()` method that currently
+displays a version of the plot in Jupyter notebooks. This works for the most
+part and is relatively simple due to Jupyter's data model. However, this
+reliance on the notebook fails for users who work primarily from the command
+line, either in the vanilla python interpreter or the IPython command line
+application. We receive many requests from confused users who do not understand
+why `show()` errors out in the regular python interpreter or appears to do
+nothing in `IPython`, when they expect a GUI window to pop up when they run
+`show()`.
+
+This project would have a student modify yt's plotting objects to hook into
+matplotlib's interactive backends so that plots can be optionally displayed
+using a GUI plotting window. Optimally, we would also enable callbacks so that
+zooming and selecting does the "right thing", generating high resolution data
+when it is available.
+
+This is constrained by maintaining backward compatibility: by default yt should
+not fail when generating plots on headless devices (e.g. when connecting over
+SSH to a supercomputer).
+
+*Deliverables:*
+
+* A proof of concept demonstrating how to hook into matplotlib's interactive
+  backends using the matplotlib object-oriented API, or a way to show how to
+  gracefully fall back to using pyplot instead of the object oriented API.
+
+* A YTEP describing the proposed approach for modifying yt's plotting
+  infrastructure to support matplotlib's interactive plotting backends.
+
+* The implementation for the YTEP submitted as a bitbucket pull request to the
+  main yt repository.
+
+### Testing improvements
```

Comment:
A more specific title might be better here.  Something like "Improve coverage and preformance of the yt testing infrastructure"?